src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -1171,3 +1171,44 @@ extern "C" char* SystemNative_RealPath(const char* path)
     assert(path != nullptr);
     return realpath(path, nullptr);
 }
+
+static int32_t LockUnlock(intptr_t fd, off_t offset, off_t length, bool lock)
+{
+    if (offset < 0 || length < 0) 
```

Comment:
It does the offset check, but it may or may not do the length check. From the docs:
```
POSIX.1-2001 allows (but does not require) an implementation to
       support a negative l_len value; if l_len is negative, the interval
       described by lock covers bytes l_start+l_len up to and including
       l_start-1.  This is supported by Linux since kernel versions 2.4.21
       and 2.5.49.
```